backend/src/impl/default_controllers_impl.py

```diff
@@ -150,12 +160,52 @@ def benchmarkconfigs_get(parent: Optional[str]) -> list[BenchmarkConfig]:
 def benchmark_benchmark_id_get(benchmark_id: str) -> Benchmark:
     config = BenchmarkConfig.from_dict(BenchmarkUtils.config_dict_from_id(benchmark_id))
     if config.type == "abstract":
-        return Benchmark(config, None)
-    sys_infos = BenchmarkUtils.load_sys_infos(config)
-    orig_df = BenchmarkUtils.generate_dataframe_from_sys_infos(config, sys_infos)
-    view_dfs = BenchmarkUtils.generate_view_dataframes(config, orig_df)
-    views = [BenchmarkUtils.dataframe_to_table(k, v) for k, v in view_dfs]
-    return Benchmark(config, views)
+        return Benchmark(config, None, None)
+    file_path = benchmark_id + "_benchmark.json"
+    plot_path = benchmark_id + "_plot.csv"
+    benchmark_file = open_cached_file(file_path, datetime.timedelta(days=1))
+    if not benchmark_file:
+        sys_infos = BenchmarkUtils.load_sys_infos(config)
+        orig_df = BenchmarkUtils.generate_dataframe_from_sys_infos(config, sys_infos)
+        view_dfs = BenchmarkUtils.generate_view_dataframes(config, orig_df)
+        json_dict = {k: v.to_dict() for k, v in view_dfs}
+        benchmark_file = os.path.join(get_cache_dir(), sanitize_path(file_path))
+        with open(benchmark_file, "w") as outfile:
+            json.dump(json_dict, outfile)
+        if config.parent:
+            list_data = [
```

Comment:
We should probably avoid explicitly writing "Demographic-weighted Global Average" and "Linguistic-weighted Global Average" for example, as it may not apply to all benchmarks. I think there's some other underlying logic that you used to pick these values as the ones you wanted to focus on, so it'd be better to have the code reflect that logic. For instance, I think you probably picked these because they only had a single column/row in the results table, right?

Maybe we could instead have the logic reflect that by using an "if" statement to check if they only have a single column.

Regarding having a single row, I think that's probably not necessary. Rather we could take the "max" over all of the rows to get the value.

Reply:
I hardcoded "Demographic-weighted Global Average" and "Linguistic-weighted Global Average" because I was under the impression that the only time we would need to plot would be for the "global benchmark"(s) because the goal of plotting would be to track the world's collective progress. Furthermore, the global benchmark json file specifies the two views ("Demographic-weighted Global Average" and "Linguistic-weighted Global Average"). I guess this could be an issue if we decide to add another global view in the global benchmark json file, but I didn't think we'd be doing that frequently. Let me know if this logic makes sense; if not, I can totally work on "unhardcoding".

Reply:
We might also want to plot the state-of-the-art on other benchmarks too. You can see an example here from PapersWithCode: https://paperswithcode.com/sota/image-classification-on-imagenet

So I think basically if any view has only one column associated with it, we could measure the progress over time for the maximum value of that column.